codegen/src/configuration.rs

```diff
@@ -0,0 +1,350 @@
+                }
+                _ => {
+                    unimplemented!("To implement.")
+                }
+            }
+        }
+    }
+
+    let references = pull_sub_schemas(&schema_object);
+
+    if let Some(references) = references {
+        push_definitions(queue, references, root_schema);
+        while let Some((_name, schema)) = queue.pop_front() {
+            top_level_object(queue, buffer, root_schema, schema, header, title.clone())?
+        }
+    }
+
+    Ok(())
+}
+
+fn push_definitions<'q, 'a>(
+    queue: &'q mut Queue<'a>,
+    references: Vec<String>,
+    root_schema: &'a RootSchema,
+) {
+    for reference in references {
+        let key = reference.trim_start_matches("#/definitions/");
+        match root_schema.definitions.get(key) {
+            Some(Schema::Object(schema)) => {
+                queue.push_back(key.to_string(), schema);
+            }
+            None => {}
+            _ => {
+                unimplemented!("To implement.")
+            }
+        }
+    }
+}
+
+fn pull_sub_schemas(schema_object: &SchemaObject) -> Option<Vec<String>> {
+    let result: Option<Vec<_>> = schema_object
+        .subschemas
+        .as_ref()
+        .and_then(|validation| validation.any_of.as_ref())
+        .map(|any_of| {
+            let res = any_of
+                .into_iter()
+                .filter(|any_of| any_of.is_ref())
+                .map(|any_of| any_of.clone().into_object())
+                .map(|schema| schema.reference.unwrap_or_default())
+                .collect();
+
+            res
+        });
+
+    result
+}
+
+pub fn generate_buffer(buffer: &mut Vec<u8>) -> anyhow::Result<()> {
+    let root = schema_for!(PartialConfiguration);
+    writeln!(buffer, "{}", generate_markdown_hearer())?;
+
+    writeln!(
+        buffer,
+        "{}",
+        &root.schema.metadata.clone().unwrap().description.unwrap()
+    )?;
+    writeln!(buffer)?;
+
+    if let Some(object) = &root.schema.object {
+        let iter = object.properties.iter().enumerate();
+        for (_index, (property_name, property_schema)) in iter {
+            let mut queue = Queue::default();
+            let schema_object = property_schema.clone().into_object();
+            writeln!(buffer, "## `{property_name}`")?;
+            writeln!(buffer)?;
+            let mut header = Header::default();
+            let title = Title {
+                prefix: property_name.to_string(),
+            };
+            top_level_object(
+                &mut queue,
+                buffer,
+                &root,
+                &schema_object,
+                &mut header,
+                title,
+            )?;
+        }
+    }
+
+    Ok(())
+}
+
+pub fn generate_configuration() -> anyhow::Result<()> {
+    let mut buffer = Vec::new();
+
+    generate_buffer(&mut buffer)?;
+
+    fs::write("src/content/docs/reference/configuration_v2.mdx", buffer)?;
+
+    Ok(())
+}
+
+fn generate_markdown_hearer() -> String {
```

Comment:
i noticed a typo in my original [function name](https://github.com/biomejs/biome/pull/1028/files#diff-bfd7157fe097ef23de66a65587759406e2ad7105b057b3915d097feb3565bd49R10)... :pray:  It should be `generate_markdown_header()` and sorry for the mistake.